invenio_records_resources/services/records/results.py

```diff
@@ -292,10 +294,12 @@ def add_service_value(self, service, value):
 
     def add_dereferenced_record(self, service, value, resolved_rec):
         """Save the dereferenced record."""
+        # mark the record as a "ghost" or "system" record i.e not resolvable
         if resolved_rec is None:
-            resolved_rec = self.ghost_record({"id": value})
-            # mark the record as a "ghost" record i.e not resolvable
-            resolved_rec["is_ghost"] = True
+            if value == system_user_id:
+                resolved_rec = self.system_record()
```

Comment:
where does this `system_record` come from?

Reply:
Yes, but then all the classes that inherit from will have to implement the `system_record()`, which doesn't make sense for them (for example `ParentCommunitiesExpandableField`)

Reply:
Then, can `self.system_record()` be undefined at some point?
You can also declare it without being abstract and it doesn't do anything (no-op). It's then overriden by whoever needs to be override it.

Reply:
Good point! Feels like better to make it fail here with the  `NotImplementedError` (same as the `ghost_record ` above, but make it not abstract), so that we know where and why it failed, WDYT?
ping @kpsherva

Reply:
👍  on following the ghost